system-test/cluster.ts

```diff
@@ -0,0 +1,313 @@
+// limitations under the License.
+
+import {beforeEach, describe, it} from 'mocha';
+import {generateId} from './common';
+import {
+  Bigtable,
+  ClusterInfo,
+  Instance,
+  Cluster,
+  GetClusterMetadataResponse,
+} from '../src';
+import assert = require('assert');
+import {ClusterUtils} from '../src/utils/cluster';
+
+describe('Cluster', () => {
+  const bigtable = new Bigtable();
+  let instance: Instance;
+
+  async function createNewInstance(clusters: ClusterInfo[]): Promise<void> {
+    const instanceId: string = generateId('instance');
+    instance = bigtable.instance(instanceId);
+    const [, operation] = await instance.create({
+      clusters,
+      labels: {
+        time_created: Date.now(),
+      },
+    });
+    await operation.promise();
+  }
+  async function createStandardNewInstance(
+    clusterId: string,
+    nodes: number
+  ): Promise<void> {
+    return await createNewInstance(standardCreationClusters(clusterId, nodes));
+  }
+  function standardCreationClusters(
+    clusterId: string,
+    nodes: number
+  ): ClusterInfo[] {
+    return [
+      {
+        id: clusterId,
+        location: 'us-east1-c',
+        nodes,
+      },
+    ];
+  }
+  afterEach(async () => {
+    await instance.delete();
+  });
+  describe('Create cluster', () => {
+    describe('With manual scaling', () => {
+      let clusterId: string;
+      let cluster: Cluster;
+      async function checkMetadata(localCluster: Cluster, nodes: number) {
+        const metadata: GetClusterMetadataResponse =
+          await localCluster.getMetadata({});
+        assert.strictEqual(metadata[0].serveNodes, nodes);
+      }
+      beforeEach(async () => {
+        clusterId = generateId('cluster');
+        await createStandardNewInstance(clusterId, 2);
+        cluster = instance.cluster(clusterId);
+      });
+      it('Create an instance with clusters for manual scaling', async () => {
+        await checkMetadata(cluster, 2);
+      });
+      it('Create an instance and then create a cluster for manual scaling', async () => {
+        const clusterId2: string = generateId('cluster');
+        const cluster2 = instance.cluster(clusterId2);
+        await cluster2.create({
+          location: 'us-west1-c',
+          nodes: 3,
+        });
+        await checkMetadata(cluster2, 3);
+      });
+      describe('Using an incorrect configuration', () => {
+        let cluster2: Cluster;
+        beforeEach(async () => {
+          const clusterId2: string = generateId('cluster');
+          cluster2 = instance.cluster(clusterId2);
+        });
+        it('Without providing any cluster configuration', async () => {
+          try {
+            await cluster2.create({
+              location: 'us-west1-c',
+            });
+            assert.fail();
+          } catch (e) {
+            assert.equal(e.message, ClusterUtils.noConfigError);
+          }
+        });
+        it('By providing too much cluster configurations', async () => {
```

Comment:
nit - reword to express that the user entered manual AND autoscaling configurations